src/Http/Http.Extensions/gen/StaticRouteHandlerModel/StaticRouteHandlerModel.Emitter.cs

```diff
@@ -205,7 +205,7 @@ public static string EmitFilteredArgumentList(this Endpoint endpoint)
 
         for (var i = 0; i < endpoint.Parameters.Length; i++)
         {
-            sb.Append($"ic.GetArgument<{endpoint.Parameters[i].Type.ToDisplayString(EmitterConstants.DisplayFormat)}>({i})");
+            sb.Append($"ic.GetArgument<{endpoint.Parameters[i].Type.ToDisplayString(EmitterConstants.DisplayFormat)}>({i})!");
```

Comment:
Hmmmm...do we want to apply the nullability suppression operator unconditionally? If the handler has a signature like:

```
app.MapGet("/hello/{name}", (string? name) => ...);
```

We run the risk of making the following invocation:

```
handler(ic.GetArgument<string>(null!))
```

If the user visits `/hello`.

I believe we'd only want to apply the `!` if the parameter is not optional...

Reply:
As discussed, in this context it makes sense to have the `!` operator. If we were to emit some branching code here that calls one way or the other depending on the nullability of the type we'd end up with effectively the same call to `GetArgument(...)` ... at which point its just added some complexity and not a lot of value.